Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/AppBar/AppBar.cs

```diff
@@ -0,0 +1,494 @@
+
+                case ButtonTypeEnum.Adjust:
+                    if (!useAdjust)
+                        return false;
+                    break;
+            }
+
+            switch (state)
+            {
+                case AppBarStateEnum.Default:
+                default:
+                    switch (buttonType)
+                    {
+                        // Show hide, adjust, remove buttons
+                        // The rest are hidden
+                        case AppBar.ButtonTypeEnum.Hide:
+                        case AppBar.ButtonTypeEnum.Remove:
+                        case AppBar.ButtonTypeEnum.Adjust:
+                        case AppBar.ButtonTypeEnum.Custom:
+                            return true;
+
+                        default:
+                            return false;
+                    }
+
+                case AppBarStateEnum.Hidden:
+                    switch (buttonType)
+                    {
+                        // Show show button
+                        // The rest are hidden
+                        case ButtonTypeEnum.Show:
+                            return true;
+
+                        default:
+                            return false;
+                    }
+
+                case AppBarStateEnum.Manipulation:
+                    switch (buttonType)
+                    {
+                        // Show done button
+                        // The rest are hidden
+                        case AppBar.ButtonTypeEnum.Done:
+                            return true;
+
+                        default:
+                            return false;
+                    }
+
+            }
+        }
+
+        public void GetButtonTextAndIconFromType(ButtonTypeEnum type, out string buttonText, out Texture buttonIcon, out int displayOrder)
+        {
+            switch (type)
+            {
+                case ButtonTypeEnum.Show:
+                    buttonText = "Show";
+                    buttonIcon = showIcon;
+                    displayOrder = 0;
+                    break;
+
+                case ButtonTypeEnum.Hide:
+                    buttonText = "Hide";
+                    buttonIcon = hideIcon;
+                    displayOrder = 1;
+                    break;
+
+                case ButtonTypeEnum.Adjust:
+                    buttonText = "Adjust";
+                    buttonIcon = adjustIcon;
+                    displayOrder = 2;
+                    break;
+
+                case ButtonTypeEnum.Remove:
+                    buttonText = "Remove";
+                    buttonIcon = removeIcon;
+                    displayOrder = 3;
+                    break;
+
+                case ButtonTypeEnum.Done:
+                    buttonText = "Done";
+                    buttonIcon = doneIcon;
+                    displayOrder = 4;
+                    break;
+
+                default:
+                    throw new ArgumentOutOfRangeException("type", type, null);
```

Comment:
nit: feels like this should be an invalid argument exception not an argument out of range exception